test/layout/test_teleporters.py

```diff
@@ -61,6 +65,66 @@ def _a(region, area, instance_id=None):
     return AreaIdentifier(region, area).as_json
 
 
+def _g(
+    encoded: bytes,
+    bit_count: int,
+    description: str,
+    num_valid_targets: int,
+    mode: str = "vanilla",
+    excluded_teleporters=None,
+    excluded_targets=None,
+):
+    if excluded_teleporters is None:
+        excluded_teleporters = []
+    if excluded_targets is None:
+        excluded_targets = []
+    return {
+        "encoded": encoded,
+        "bit_count": bit_count,
+        "description": description,
+        "num_valid_targets": num_valid_targets,
+        "json": {
+            "mode": mode,
+            "excluded_teleporters": excluded_teleporters,
+            "excluded_targets": excluded_targets,
+        },
+    }
+
+
+@pytest.fixture(
+    params=[
+        _g(b"\x00", 3, "Original connections", 40),
+        _g(b"\xc0", 6, "One-way, with cycles", 40, mode="one-way-teleporter"),
+        _g(
+            b"\xd8\x12",
+            16,
+            "One-way, with replacement; excluded 1 elevators",
+            40,
+            mode="one-way-teleporter-replacement",
+            excluded_teleporters=[
+                _a("Area 1", "Transport to Surface and Area 2", "Elevator to Surface East"),
+                _a("Area 4 Crystal Mines", "Transport to Area 4", "Elevator to Central Caves"),
```

Comment:
Is this intentionally using the wrong area and node name resulting in one excluded teleporter, or is this test case supposed to exclude 2 teleporters?

Reply:
It was supposed to be two but the test kept throwing an error saying it was 1. Not sure if I just did something wrong there.

Reply:
I think that makes sense with what I was expecting. I'm fixing it in #8426